server/reload.go

```diff
@@ -663,26 +687,65 @@ func (s *Server) applyOptions(opts []option) {
 func (s *Server) reloadAuthorization() {
 	s.mu.Lock()
 	s.configureAuthorization()
+	oldAccounts := s.accounts
+	gAccount := oldAccounts[globalAccountName]
+	s.accounts = make(map[string]*Account)
+	s.registerAccount(gAccount)
+	for _, newAcc := range s.opts.Accounts {
+		// For existing accounts, moved updated config to existing account
+		if acc, ok := oldAccounts[newAcc.Name]; ok {
+			acc.mu.RLock()
+			sl := acc.sl
+			acc.mu.RUnlock()
+			newAcc.mu.Lock()
+			newAcc.sl = sl
+			// Check if current and new config of this account are same
+			// in term of stream imports.
+			newAcc.checksti = !acc.checkStreamImportsEqual(newAcc)
+			newAcc.mu.Unlock()
+		}
+		s.registerAccount(newAcc)
+	}
+	// Gather clients that changed accounts. We will close them and they
+	// will reconnect, doing the right thing.
+	closeClients := make(map[uint64]*client, len(s.clients))
 	clients := make(map[uint64]*client, len(s.clients))
 	for i, client := range s.clients {
-		clients[i] = client
+		if s.clientHasMovedToDifferentAccount(client) {
+			closeClients[i] = client
```

Comment:
This can leave nil holes in each array.

Reply:
Probably best to do appends to each instead of direct indexing.

Reply:
I think its ok, but if you want to change go for it, no reason for maps AFAIK.

Reply:
Changing it to array. Was there as map initially with clients and continued to use that, but true that it would not have been my first choice.